### ISSUE TITLE
Enable "Remember password" option back

### ISSUE THREAD

**Issue by POMATu** [RetroShare/RetroShare#1754]
I am using RS 0.6.5-retroshare-gui-OBS-deb. "Remember password" option is gone and in settings its disabled.

So people on chatserver said that I need to add compile flags to enable this feature. You had the option previously, why u had disabled it? (this is a rhetorical question I know that its not secure). To make it work like I want now I need to recompile RS on every update, while you offer convenient OBS repo. And I wanna run 24/7 AAM node that requires autologin functionality.

If you are concerned about security you can just put this option under multiple warnings.

I don't like when software I use thinks it's smarter than me. Looks like we are going w1nd0w$ way mates.

**Comment by POMATu** [comment]
Or maybe there is another solution? What happens if I decrypt PGP cert locally? Will I get autologin functionality then without password popup on startup?

**Comment by G10h4ck** [comment]
The problem is not just security, autologin was implemented on top of an old deprecated library, that now is not available anymore on most linux distributions, so we cannot package it with autologin enabled until someone rewrite it from scratch using a modern library,

About having PGP cert unencrypted you could do that manipulating the certificate with gpg 1.x but it seems a bit risky.

What I suggest for your node is to use JSON API to automatically login via a script like this:

`curl --data '{"account":"160bfc8fcf0cdb00000fffffb40c60b","password":"01234_bum"}' http://127.0.0.1:9092/rsLoginHelper/attemptLogin`

This should work like a charm at least with `retroshare-service`

**Comment by POMATu** [comment]
Hi,
So it might be a solution to the problem, however my JSON API dont work at all
```
curl: (7) Failed to connect to 127.0.0.1 port 9092: Connection refused
```

Even tho I have enabled json api in settings

**Comment by POMATu** [comment]
yh look so json api dont work on RS login dialog, it works only after splash screen, so when RS being loaded completely

**Comment by georgedorn** [comment]
Yeah, I've also tested this and found that the json API is not enabled until after login, so it can't be used to log in.

Are there any other mechanisms for automatic login?  I'm trying to run a relay on a server with limited memory, and thanks to memory leaks it needs to be restarted daily.

**Comment by G10h4ck** [comment]
> I'm trying to run a relay on a server with limited memory, and thanks to memory leaks it needs to be restarted daily.

On a server I would use retroshare-service which doesn't have GUI, uses less RAM, and have JSON API enabled since it starts (before login) so you can use it to automatically login, if your profile is already created you can use it with retroshare-service because it uses same identical format then retroshare-gui

**Comment by georgedorn** [comment]
> On a server I would use retroshare-service which doesn't have GUI, uses less RAM, and have JSON API enabled since it starts (before login) so you can use it to automatically login

The last time I looked into this, there was no working client for the non-gui version.  There used to be one, but that was ages ago and it doesn't work anymore.  Has this changed?

**Comment by G10h4ck** [comment]
The old retroshare-nogui has been removed, and replaced with retroshare-service which is completely controllable via JSON API, as of today I am tempted to say it works better then retroshare-gui.

**Comment by POMATu** [comment]
So issue not solved for retroshare-gui yet? JSON api still dont come before login?

**Comment by G10h4ck** [comment]
> So issue not solved for retroshare-gui yet? JSON api still dont come before login?

I think it should not be terribly difficult to turn on JSON API before login on retroshare-gui either but @csoler sure have more insight then me I avoid touching retroshare-gui code as much as I can, while I code gladly in libretroshare and retroshare-service

**Comment by POMATu** [comment]
so few years later (actually almost 4) this is still not fixed but i delivered sorta stable workaround for retroshare-gui autologin

i left it here:
https://github.com/POMATu/retroshare-autologin

enjoy

**Comment by defnax** [comment]
@csoler @thunder2 close
